Github Actions: Windows Artifact Upload Failing

### Operating System

N/A

### OpenRCT2 build

N/A

### Base game

RollerCoaster Tycoon 2, RollerCoaster Tycoon Classic

### Area(s) with this issue?

Issues building the game

### Describe the issue

There have been no Windows artifacts for the past couple of development builds, with `v0.4.3-e6a0262` being the last to have any windows artifacts. Example: https://openrct2.org/downloads/develop/v0.4.3-a982d15

Looking at the GitHub action it looks like all Windows jobs are failing to upload the artifacts. Example: https://github.com/OpenRCT2/OpenRCT2/actions/runs/4470488861/jobs/7854130448

```
Setting up environment for OpenRCT2...
Uploading to openrct2.org as 'OpenRCT2-0.4.3-develop-6170bb0-windows-portable-x64.zip'...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0 37.1M    0     0    0 65535      0   310k  0:02:02 --:--:--  0:02:02  312k
curl: (55) Failed sending data to the peer
Error: Process completed with exit code 55.
```

### Steps to reproduce

N/A - not specific to game. Build infrastructure.

### Attachments

_No response_

--- a/scripts/upload-build
+++ b/scripts/upload-build
@@ -33,7 +33,11 @@ if [ -z "$OPENRCT2_ORG_TOKEN" ]; then
     echo -e "\033[0;31mOPENRCT2_ORG_TOKEN not set"
     exit 1
 fi
+# Use HTTP1.1 as a workaround for curl bug:
+# https://github.com/actions/runner-images/issues/7329
+# https://github.com/curl/curl/issues/10591
 curl --include -m 300 --connect-timeout 5 -o - \
+    --http1.1 \
     --form "key=$OPENRCT2_ORG_TOKEN" \
     --form "fileName=$filename" \
     --form "version=$version" \

